api/audit-trail-gw/src/services/channel-info-service.ts

```diff
@@ -1,17 +1,40 @@
-import { ChannelInfo, ChannelInfoPersistence, ChannelInfoSearch, getDateFromString, getDateStringFromDate } from '@iota/is-shared-modules';
+import { ChannelInfo, ChannelInfoPersistence, ChannelInfoSearch, ChannelType, getDateFromString, getDateStringFromDate } from '@iota/is-shared-modules';
 import * as ChannelInfoDb from '../database/channel-info';
 import isEmpty from 'lodash/isEmpty';
 
 export class ChannelInfoService {
-	async getChannelInfo(channelAddress: string): Promise<ChannelInfo | null> {
+	async getChannelInfo(channelAddress: string, reqId: string, isAdmin: boolean): Promise<ChannelInfo | null> {
 		const channelInfoPersistence = await ChannelInfoDb.getChannelInfo(channelAddress);
-		return channelInfoPersistence && this.getChannelInfoObject(channelInfoPersistence);
+		const channelInfo = channelInfoPersistence && this.getChannelInfoObject(channelInfoPersistence);
+		const authorId = channelInfo?.authorId;
+		const hidden = channelInfo?.hidden;
+		if (hidden && !isAdmin && reqId !== authorId) {
+			throw new Error('not allowed for hidden channels!');
+		}
+		if (!isAdmin && reqId !== authorId && !hidden) {
+			channelInfo.visibilityList = [];
+		}
+		return channelInfo;
 	}
 
-	async searchChannelInfo(channelInfoSearch: ChannelInfoSearch): Promise<ChannelInfo[]> {
+	async searchChannelInfo(channelInfoSearch: ChannelInfoSearch, reqId: string, isAdmin: boolean): Promise<ChannelInfo[]> {
 		let channelInfoPersistence: ChannelInfoPersistence[] = [];
 		channelInfoPersistence = await ChannelInfoDb.searchChannelInfo(channelInfoSearch);
-		return channelInfoPersistence?.map((channel) => this.getChannelInfoObject(channel)).filter((c) => c);
+		let channelInfos = channelInfoPersistence?.map((channel) => this.getChannelInfoObject(channel)).filter((c) => c);
+		channelInfos = channelInfos
+			.filter(ch => {
+				const isAuthor = reqId === ch.authorId
+				const visbilityListSearch = channelInfoSearch.visibilityList
+				return (!isAdmin && !isAuthor && !ch?.hidden && !visbilityListSearch) || (isAdmin || isAuthor)
+			})
+			.map(ch => {
+				const isAuthor = reqId === ch.authorId
```

Comment:
why is this map necessary?

Reply:
I think it is same as above in getchannelinfo. setting the list to empty array is not necessary

Reply:
ok sounds like michele wants to have it..